### PR TITLE
Room controller tests

### PR DESCRIPTION
**Pull request by illeanz**
sorry for the mess!

They all pass currently. Please let me know if there's anything I missed! Kinda-sleepy brain is even worse than regular brain.

### CODE REVIEW

File: uwuno/src/test/java/com/learning/uwuno/unit/roomControllerTests.java [hunk 1]
```diff
@@ -28,10 +31,191 @@
...
+                .andExpect(status().isOk())
+                .andExpect(jsonPath("$[0]['name']").value("test"))
+                .andExpect(jsonPath("$.length()").value(1))
+                .andDo(print());
+    }
+
+    @Test
+    public void GET_room() throws Exception {
+        room room = new room("test", false);
+        when(service.getRoom("2")).thenReturn(room);
```
Comment (rico-law) [inline uwuno/src/test/java/com/learning/uwuno/unit/roomControllerTests.java]: Can do anyString() instead of a hard coded "2", so it will work in case someone changes the 2 later down for accident.

Comment (illeanz) [reply uwuno/src/test/java/com/learning/uwuno/unit/roomControllerTests.java]: Yessir! I'll change it 👍👍

File: uwuno/src/test/java/com/learning/uwuno/unit/roomControllerTests.java [hunk 2]
```diff
@@ -28,10 +31,191 @@
...
+                .andExpect(status().isOk())
+                .andExpect(content().contentType(MediaType.APPLICATION_JSON))
+                .andExpect(jsonPath("$.name").value(room.getName()))
+                .andExpect(jsonPath("$.uid").value(room.getUid()))
+                .andDo(print());
+    }
+
+    @Test
+    public void POST_add_room_bad_req_err() throws Exception {
+        when(service.addRoom(isNull(), anyBoolean())).thenThrow(new com.learning.uwuno.errors.badRequest());
```
Comment (rico-law) [inline uwuno/src/test/java/com/learning/uwuno/unit/roomControllerTests.java]: I think we should be testing a bad input JSON instead of just throwing in the function and then checking if the function you mocked threw. That's testing their when().thenThrow() not our functions.
I think this applies in some of the other error tests too, for the tests checking for correctness mocking a object return is fine because I think it will help test the serialization is working correctly?

**Review by rico-law — CHANGES_REQUESTED** [review]
Some more small changes

File: uwuno/src/test/java/com/learning/uwuno/unit/roomControllerTests.java (hunk 2 above)
Comment (illeanz) [reply uwuno/src/test/java/com/learning/uwuno/unit/roomControllerTests.java]: mmm that's a good point. I missed that... I'll add some more tests zzz

File: uwuno/src/test/java/com/learning/uwuno/unit/roomControllerTests.java [hunk 3]
```diff
@@ -28,10 +31,231 @@
...
+                .accept(MediaType.APPLICATION_JSON))
+                .andExpect(status().isInternalServerError())
+                .andDo(print());
+    }
 
     @Test
-    public void testRoom_POST() throws Exception {
-//        this.mock.perform(post("/Rooms")).andDo(print()).andExpect(status().isOk());
+    public void POST_add_room_bad_req_err() throws Exception {
+        when(service.addRoom(anyString(), anyBoolean())).thenThrow(new com.learning.uwuno.errors.badRequest());
```
Comment (rico-law) [inline uwuno/src/test/java/com/learning/uwuno/unit/roomControllerTests.java]: You're not testing anything here. No matter error or no error, you're gunna get a bad request.
Recommend: Remove all thenThrow() + test for throws. The controller should throw if the wrong parameters are passed in to start with.

Comment (illeanz) [reply uwuno/src/test/java/com/learning/uwuno/unit/roomControllerTests.java]: It depends on the function. This test suite is designed to test the room controller, not the service. If the controller's function is just calling the service and catching the error, this test makes sense. I changed the test for the POST room function because there are actual checks within the controller function before calling service functions that would throw an error.

File: uwuno/src/test/java/com/learning/uwuno/unit/gameServiceTest.java [hunk 4]
```diff
@@ -52,7 +52,7 @@ public void testGameService_addRoom() throws Exception {
     }
 
     @Test
-    public void testGameService_addPlayer() throws Exception {
+    public void testGameService_addPlayer() {
         room room = gameService.addRoom(testRoomName, false);
         assertThrows(NoSuchElementException.class, () -> {
```
Comment (rico-law) [inline uwuno/src/test/java/com/learning/uwuno/unit/gameServiceTest.java]: Is it possible you did this on a not too recent branch, I'm pretty sure all the service functions were changed to throw our own excetpions.

File: uwuno/src/main/java/com/learning/uwuno/services/gameService.java [hunk 5]
```diff
@@ -23,20 +23,15 @@
...
     }
 
-    public String addRoom(room newRoom) {
-        roomList.add(newRoom);
-        return newRoom.getUid();
-    }
-
     public player addPlayer(String name, String uid) {
-        if (name.isEmpty())
+        if (name.isBlank())
```
Comment (rico-law) [inline uwuno/src/main/java/com/learning/uwuno/services/gameService.java:33]: This change is also in my branch already.
I think you forgot to merge my current branch into yours before starting the fixes.

Comment (rico-law) [reply uwuno/src/main/java/com/learning/uwuno/services/gameService.java:33]: I just merged mine into master, will have some conflicts you'll need to fix before testing everything again :(

File: uwuno/src/test/java/com/learning/uwuno/unit/gameServiceTest.java (hunk 4 above)
Comment (achang1) [reply uwuno/src/test/java/com/learning/uwuno/unit/gameServiceTest.java]: It was merged from the old master

File: uwuno/src/main/java/com/learning/uwuno/services/gameService.java (hunk 5 above)
Comment (achang1) [reply uwuno/src/main/java/com/learning/uwuno/services/gameService.java:33]: Yup. Done.

File: uwuno/src/main/java/com/learning/uwuno/services/gameService.java [hunk 6]
```diff
@@ -17,7 +17,7 @@
     // Singleton for holding anything we need everywhere, should handle all business logic
 
     // Variables
-    private ArrayList<room> roomList = new ArrayList<room>();
+    private final ArrayList<room> roomList = new ArrayList<>();
```
Comment (rico-law) [inline uwuno/src/main/java/com/learning/uwuno/services/gameService.java:20]: Why was this changed to final, if I understand correctly final == c++ const, and we definitely need to change this over the lifetime of the application.

File: uwuno/src/test/java/com/learning/uwuno/integration/tests/basicRoomTest.java [hunk 7]
```diff
@@ -15,7 +15,7 @@
 
 @TestInstance(TestInstance.Lifecycle.PER_CLASS)
 public class basicRoomTest {
-    private static ArrayList<String> runningRoomIds = new ArrayList<String>();
+    private final static ArrayList<String> runningRoomIds = new ArrayList<>();
```
Comment (rico-law) [inline uwuno/src/test/java/com/learning/uwuno/integration/tests/basicRoomTest.java:18]: What is this array used for, I couldn't find anything with a search :(

File: uwuno/src/test/java/com/learning/uwuno/unit/roomControllerTests.java [hunk 8]
```diff
@@ -1,35 +1,209 @@
...
+                .accept(MediaType.APPLICATION_JSON))
+                .andExpect(status().isOk())
+                .andExpect(jsonPath("$[0]['name']").value("testRoom"))
+                .andExpect(jsonPath("$.length()").value(1))
+                .andDo(print());
+    }
+
+    @Test
+    public void GET_room() throws Exception {
+        room.addPlayer(new player("testPlayer"));
```
Comment (rico-law) [inline uwuno/src/test/java/com/learning/uwuno/unit/roomControllerTests.java]: Maybe just create one instance of this right after room at the very beginning instead.

File: uwuno/src/test/java/com/learning/uwuno/unit/gameServiceTest.java [hunk 9]
```diff
@@ -41,7 +40,7 @@
     final private String newStatus = "Start";
 
     @Test
-    public void testGameService_addRoom() throws Exception {
+    public void testGameService_addRoom() {
```
Comment (rico-law) [inline uwuno/src/test/java/com/learning/uwuno/unit/gameServiceTest.java:43]: What's the difference between removing the "throws" here vs having them still there inside the playerController test?

File: uwuno/src/test/java/com/learning/uwuno/integration/tests/basicRoomTest.java (hunk 7 above)
Comment (achang1) [reply uwuno/src/test/java/com/learning/uwuno/integration/tests/basicRoomTest.java:18]: Nani? It's used in a few tests for clean-up (@AfterAll)

File: uwuno/src/test/java/com/learning/uwuno/unit/gameServiceTest.java (hunk 9 above)
Comment (achang1) [reply uwuno/src/test/java/com/learning/uwuno/unit/gameServiceTest.java:43]: What do you mean? It was never used

File: uwuno/src/test/java/com/learning/uwuno/integration/tests/basicRoomTest.java (hunk 7 above)
Comment (rico-law) [reply uwuno/src/test/java/com/learning/uwuno/integration/tests/basicRoomTest.java:18]: Oops I was on wrong file. But why was this changed to final? We add new rooms to it in the tests.

File: uwuno/src/test/java/com/learning/uwuno/unit/gameServiceTest.java (hunk 9 above)
Comment (rico-law) [reply uwuno/src/test/java/com/learning/uwuno/unit/gameServiceTest.java:43]: Oh I meant since the "throws" was removed here why are they inside the room tests (not the player mis-wrote). Is there a specific reason?

File: uwuno/src/test/java/com/learning/uwuno/integration/tests/basicRoomTest.java (hunk 7 above)
Comment (rico-law) [reply uwuno/src/test/java/com/learning/uwuno/integration/tests/basicRoomTest.java:18]: Oh I just read that final only prevents the reassigning of the variable not if it is changed, so I think you're right here and the other final arraylist. Sorry about that.

Comment (achang1) [reply uwuno/src/test/java/com/learning/uwuno/integration/tests/basicRoomTest.java:18]: Basically the variable can't be reassigned to something else. Intellij suggested it. It can still be modified

File: uwuno/src/test/java/com/learning/uwuno/unit/gameServiceTest.java (hunk 9 above)
Comment (achang1) [reply uwuno/src/test/java/com/learning/uwuno/unit/gameServiceTest.java:43]: Oh the other test file uses functions in the jsonUtils that needs the throw.

File: uwuno/src/test/java/com/learning/uwuno/integration/tests/basicPlayerTest.java [hunk 10]
```diff
@@ -193,7 +193,7 @@ public void putValidPlayerName200() throws FileNotFoundException {
     public void putInvalidPlayerName400() throws FileNotFoundException {
         // Player
         String playerFilePath = JSON_REQUESTS_PATH + "/postPutPlayer.json";
-        String playerInputName = "player_put_old_name_valid";;
+        String playerInputName = "player_put_old_name_valid";
```
Comment (rico-law) [inline uwuno/src/test/java/com/learning/uwuno/integration/tests/basicPlayerTest.java]: Oops didn't see this before, if this string is used in multiple places should move to the top and only create one string?

**Review by rico-law — APPROVED** [review]
